### PR TITLE
feat(codepipeline-action): add support for externalEntityLink in the manual approval action

### PR DESCRIPTION
**Pull request by michaelbrewer**
Fixes issue #5557

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/package.json [hunk 1]
```diff
@@ -151,6 +151,7 @@
       "props-default-doc:@aws-cdk/aws-codepipeline-actions.ManualApprovalActionProps.additionalInformation",
       "props-default-doc:@aws-cdk/aws-codepipeline-actions.ManualApprovalActionProps.notificationTopic",
       "props-default-doc:@aws-cdk/aws-codepipeline-actions.ManualApprovalActionProps.notifyEmails",
+      "props-default-doc:@aws-cdk/aws-codepipeline-actions.ManualApprovalActionProps.externalEntityLink",
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/package.json]: To make this warning go away, you need to provide documentation for the default value of the property, like this:

```ts
  /**
   * URL you want to provide to the reviewer as part of the approval request.
   * The URL must begin with 'http://' or 'https://'.
   *
   * @default the approval request will not have an external link
   */
  readonly externalEntityLink?: string;
```

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts [hunk 2]
```diff
@@ -24,6 +24,11 @@ export interface ManualApprovalActionProps extends codepipeline.CommonAwsActionP
    * Any additional information that you want to include in the notification email message.
    */
   readonly additionalInformation?: string;
+
+  /**
+   * URL you want to provide to the reviewer as part of the approval request. The URL must begin with 'http://' or 'https://'.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts]: Is that condition ("must begin with 'http://' or 'https://'") checked by the CodePipeline API? Should we add some validation on our side that errors out if that is not the case?

File: packages/@aws-cdk/aws-codepipeline-actions/package.json (hunk 1 above)
Comment (michaelbrewer) [reply packages/@aws-cdk/aws-codepipeline-actions/package.json]: thanks

File: packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts (hunk 2 above)
Comment (michaelbrewer) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts]: @skinny85  
I am not sure, i know that the AWS console does this kind of validation. Are there some common examples of this kind of validation in CDK?

Comment (michaelbrewer) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts]: @skinny85  - Would something like this work?

```
    if (this.props.externalEntityLink) {
      const pattern = /^((http|https):\/\/)/;
      if (!pattern.test(this.props.externalEntityLink)) {
        throw Error("'externalEntityLink' - The URL must begin with 'http://' or 'https://'");
      }
    }
```

Comment (skinny85) [reply packages/@aws-cdk/aws-codepipeline-actions/lib/manual-approval-action.ts]: Yep, that could work :) (+ a unit test for this behavior, of course 😃). _However_, I'd rather not add this validation unless the CodePipeline API performs it as well (I don't think we should disallows things that the API accepts). Can you verify whether the CodePipeline API fails if you provide something that does not start with 'http://' or 'https://'? Thanks!

**Comment by skinny85** [comment]
> > @michaelbrewer any updates on that server-side validation?
> > I'm requesting changes so that this is removed from my To-Do list, let me know, and I'll get back to it!
> 
> APL does allow for urls without http:// or https://, so i updated the docs to reflect this.

Sorry, by "APL", do you mean the CodePipeline API?

**Comment by michaelbrewer** [comment]
> CodePipeline API?

Yes, i tried sending through a url without https:// or http:// and the Codepipeline API allowed for it.

**Comment by skinny85** [comment]
Thanks @michaelbrewer !

**Pull request merged**